lib/ansiblelint/__init__.py

```diff
@@ -129,8 +137,9 @@ def matchyaml(self, file, text):
             if not isinstance(result, list):
                 raise TypeError("{} is not a list".format(result))
 
-            for section, message in result:
-                matches.append(Match(play[ansiblelint.utils.LINE_NUMBER_KEY],
+            for section, message, *optional_linenumber in result:
```

Comment:
I think the fallback extraction doesn't belong in the helper and should be done outside of the loop to avoid multiple access operations:
```suggestion
            fallback_line_number = play[ansiblelint.utils.LINE_NUMBER_KEY]
            for section, message, *optional_linenumber in result:
```